cmd/explaintest/t/join_elimination.test

```diff
@@ -0,0 +1,14 @@
+CREATE TABLE `a` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `a_id_index` (`id`)
+);
+
+CREATE TABLE `b` (
+  `x` int(11) DEFAULT NULL,
+  `y` bigint(20) DEFAULT NULL,
+  `z` int(11) DEFAULT NULL,
+  UNIQUE KEY `b_id_index` (`x`,`y`,`z`)
+);
+
+EXPLAIN SELECT a.id FROM a LEFT JOIN b ON b.z = a.id WHERE (b.y = 1 OR a.id = 20) GROUP BY a.id;
```

Comment:
Maybe we can simplify the test case further more to be like:
```
mysql> create table a(c1 int);
Query OK, 0 rows affected (0.01 sec)

mysql> create table b(c1 int);
Query OK, 0 rows affected (0.01 sec)

mysql> select distinct a.c1 from a left join b on true where (b.c1 = 1 or a.c1 = 1);
ERROR 1105 (HY000): Can't find column test.b.c1 in schema Column: [test.a.c1] Unique key: []
```
and put it in the test file?